### ISSUE TITLE
[INTERNAL CHANGE] Strongly typed `InternalEndpointRequestHandler`

### ISSUE THREAD

**Issue by PSchmiedmayer** [Apodini/Apodini#99]
**Is your change related to a problem? Please describe.**

 It would be interesting to explore if the `InternalEndpointRequestHandler` as introduced by #98 could return a typed version of `EventLoopFuture<Action<AnyEncodable>>` that has the type `endpoint.responseType` (`EventLoopFuture<Action<R>>`).
```swift
    override func callAsFunction(request exporterRequest: I.ExporterRequest, eventLoop: EventLoop) -> EventLoopFuture<Action<AnyEncodable>> {
        /* ... */
            // The current approach has a type erasured `Action` at the end
            // We actually know that the last type must be `endpoint.responseType` but we can not pass it to the `typed` 
            // method of `typedAction` as we don't have it as generic constraint.
            .map { typeErasuredAction in
                // Throws a compile error:
                typeErasuredAction.typed(endpoint.responseType.ResponseContent.self)
            }
    }
```

**Propose a solution**
We might want to consider adding an additional generic constraint to `Endpoint`.

**Comment by bauer-andreas** [comment]
I think this can be very well possible. At least for Endpoints without `ResponseTransformers` I see that perfectly possible with the current code base.  The Handler is available as a generic type, thus also the Response Type.
However not sure how we can get the return type of `ResponseTransformers` in such a generic way. We currently are completely missing type information about those.

**Comment by PSchmiedmayer** [comment]
Yes, I think that `ResponseTransformer`s are probably posing the biggest issue for this.
